### PR TITLE
Init of release

### PR DESCRIPTION
**Pull request by inverse**
- Wire up release flow (untested, any may fail)
- Add `Makefile` for packaging simplicity
- Make docker script exit 1 on wrong input (will cause CI to fail if so)

I think there could be lots of improvements still to be made. e.g. within the dockerfiles themselves and drop qemu to do multiarch builds simplier, but thats outside the scope of this.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,12 +19,43 @@ jobs:
...
+      BUILD_REF: ${{ github.sha }}
+      BUILD_VERSION: ${{ github.ref_name }}
+    steps:
+    - uses: actions/checkout@master
+    - name: Prepare
+      run: |
+        mkdir _tmp
+        ./.docker/docker.sh prepare
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
```
Comment (inverse) [inline .github/workflows/main.yml:43]: Leverages https://github.com/docker/login-action

and will need credentials in the secrets to work

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -17,12 +19,43 @@ jobs:
...
+      run: ./.docker/docker.sh build
+    - name: Test docker images
+      run: ./.docker/docker.sh test
+    - name: Push all docker images
+      run: ./.docker/docker.sh push
+    - name: Create and push manifest list
+      run: ./.docker/docker.sh manifest-list
+    - name: Package
+      run: make package
+    - name: Create GitHub release
```
Comment (inverse) [inline .github/workflows/main.yml:57]: Could be extended to increase release body and modify title etc.

Right now the body will be empty of the GitHub release and the title will match the tag


See: https://github.com/softprops/action-gh-release

**Comment by inverse** [comment]
@reloxx13 when merging Pull Requests you can use the Squash function to collapse all the commits inbetween into a single one to get cleaner history.

image


This can be made the default in the repo settings :)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -17,12 +19,43 @@ jobs:
...
-        mkdir tmp
+        mkdir _tmp
         ./.docker/docker.sh prepare
     - name: Build docker images
       run: ./.docker/docker.sh build
     - name: Test docker images
       run: ./.docker/docker.sh test
+  release:
+    runs-on: ubuntu-20.04
+    if: ${{ github.ref_type == 'tag' }}
```
Comment (inverse) [inline .github/workflows/main.yml:32]: To block running on normal code paths

**Comment by inverse** [comment]
@reloxx13 had a chance to look at this?

**Review by reloxx13 — APPROVED** [review]

**Comment by reloxx13** [comment]
is it safe to merge?

File: .github/workflows/main.yml (hunk 1 above)
Comment (inverse) [reply .github/workflows/main.yml:43]: @reloxx13 for this to work we need to include the dockerhub username/token in the repo secrets

**Pull request merged**